### PR TITLE
refactor to handle virtual workers too

### PR DESCRIPTION
**Pull request by karmab**
- rename virtual parameters to virtual_masters_*
- added virtual_workers* parameters
- remove the metal3-config stuff which was from 4.3

### CODE REVIEW

File: 05_cache.sh [hunk 1]
```diff
@@ -18,13 +18,3 @@ SPACES=$(grep apiVIP /root/install-config.yaml | sed 's/apiVIP.*//' | sed 's/ /\
...
-
-# only needed for 4.3
-cd /root
-if [ -z "$COMMIT_ID" ] ; then
-export COMMIT_ID=$(openshift-baremetal-install version | grep '^built from commit' | awk '{print $4}')
-fi
-export RHCOS_URI=$(curl -s -S https://raw.githubusercontent.com/openshift/installer/$COMMIT_ID/data/data/rhcos.json | jq .images.openstack.path | sed 's/"//g')
-export RHCOS_PATH=$(curl -s -S https://raw.githubusercontent.com/openshift/installer/$COMMIT_ID/data/data/rhcos.json | jq .baseURI | sed 's/"//g')
-export PRIMARY_IP=$(hostname -I | cut -d" " -f1)
-envsubst < metal3-config.yaml.sample > metal3-config.yaml
```
Comment (yuvalk) [inline 05_cache.sh:30]: how is this related to virtual workers?

File: extra_worker.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  manager: baremetal-operator
+  name: {{ cluster }}-worker-1-bmc-secret
+  namespace: openshift-machine-api
+type: Opaque
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: {{ cluster }}-worker-1
```
Comment (yuvalk) [inline extra_worker.yml:15]: is this an example? cause what happen if more than one? need to be worker-2, etc.

File: lab.yml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 lab: true
 config_host: 192.168.129.1
-virtual: true
+virtual_masters: true
+virtual_workers: true
 deploy: false
 cluster: lab
-domain: cnftigerteam.com 
+domain: karmalabs.com
```
Comment (yuvalk) [inline lab.yml:7]: no more tigers? :-)

File: 05_cache.sh (hunk 1 above)
Comment (karmab) [reply 05_cache.sh:30]: not at all, but i ve decided it was time to remove this 4.3 stuff

File: lab.yml (hunk 3 above)
Comment (karmab) [reply lab.yml:7]: i came to realize i was not a tiger

File: extra_worker.yml (hunk 2 above)
Comment (karmab) [reply extra_worker.yml:15]: there is a flag, virtual_workers_deploy set to False by default, so that you can actually deploy manually your worker after initial install time.
this is there only for lab purposes, since i believe it could be good to do this manual worker thing as an extra lab step
